storage/innobase/log/log0recv.cc

```diff
@@ -4855,19 +4873,21 @@ byte *recv_dblwr_t::find_page(const page_id_t page_id,
     }
 
     const lsn_t lsn= mach_read_from_8(page + FIL_PAGE_LSN);
-    if (lsn <= max_lsn ||
-        !validate_page(page_id, page, space, tmp_buf))
+    if (lsn > min_lsn ||
+        lsn < recv_sys.parse_start_lsn ||
+        !validate_page(page_id, page, space, tmp_buf, true))
     {
       /* Mark processed for subsequent iterations in buf_dblwr_t::recover() */
-      memset(page + FIL_PAGE_LSN, 0, 8);
+      memset_aligned<8>(page + FIL_PAGE_LSN, 0, 8);
       continue;
     }
 
-    ut_a(page_get_page_no(page) == page_id.page_no());
-    max_lsn= lsn;
+    min_lsn= lsn;
     result= page;
   }
 
+  if (result)
+    recv_sys.check_page_lsn(min_lsn);
```

Comment:
bdfcd5c148b65e9073f41e861b1e16943d7abcad revises this so that the initial value of `min_lsn` will be the current LSN. In that way, we will disregard any doublewrite pages that might be in the future. Currently it is also skipping the assignment of `recv_sys.max_page_lsn`.

I’m starting to think that the variable `recv_sys.max_page_lsn` is not serving any useful purpose, because whenever we would updated it, we would already have checked that the `FIL_PAGE_LSN` is not in the future.

Reply:
3000bbd35d4955b2cfc0d7dfb2221234d894e126 removes `recv_max_page_lsn` and `recv_lsn_checks_on` without any replacement.